masking_package/background_functions.py

```diff
@@ -0,0 +1,149 @@
+	if task_local:
+		complexity += 1
+	if task_disc:
+		complexity += 1
+
+	#if neither localization or discrimination tasks occur,
+	#the number of trials does not need to be adjusted
+	#(i.e., the if-statements are skipped)
+
+	#if we have either the localization or discrimination task,
+	if complexity == 1:
+		#we only need to ensure that the number of trials is even
+		if N_entered % 2 != 0:
+			N_entered += 1
+	#if it's both of these tasks,
+	if complexity == 2:
+		#the number of trials needs to be divisible by 4 (i.e., 2x2 trial types)
+		while N_entered % 4 != 0:
+			N_entered += 1
+
+	#creating the actual list of trials (i.e, the trial sequence)
+
+	#creating a list of trial types (each number being another type of trial)
+	global N_types
+	N_types = []
+	i = 0
+	#if there is one of tasks that demand multiple trial types
+	if complexity == 1:
+		if task_local: #we check the type of task, if it's localization...
+			while i < (N_entered/2):
+				N_types.append(1) #half the trials will be on the left
+				N_types.append(2) #the other on the right
+				i += 1
+		else: #if it's the discrimination task
+			while i < (N_entered/2):
+				N_types.append(5) #half the trials will show targets (center of screen)
+				N_types.append(6) #the other half, will show foils (center of screen)
+				i += 1
+	elif complexity == 2: #if it's both
+		while i < (N_entered/4):
+				N_types.append(1) #a quarter shows target on the left
+				N_types.append(2) #another quarter shows target on the right
+				N_types.append(3) #another: foils on the left
+				N_types.append(4) #another: foils on the right
+				i += 1
+	else: #if it's neither of these 2 tasks (no task, or only detection task)
+		while i < N_entered:
+			N_types.append(5) #on all trials, stimuli are presented in the middle of the screen
+			i += 1
+
+	i = 0
+	#then, we add the blank trials (if there are any)
+	while i < N_blanks:
+		N_types.append(9)
+		i += 1
+	#And finally, randomize the order of trials
+	shuffle(N_types)
+
+
+def determine_masking(mask_type):
+	'''Translates string input from users into boolean values'''
+	global mask_forw, mask_backw
+	if mask_type in ('Forw', 'forw', 'Forward', 'forward', 'f'):
+		mask_forw = True
+	elif mask_type in ('Backw', 'backw', 'Backward', 'backward', 'b'):
+		mask_backw = True
+	elif mask_type in ('Sandw', 'sandw', 'Sandwich', 'sandwich', 's'):
+		mask_forw = True
+		mask_backw = True
+	else:
+		print(f'Warning, {mask_type} does not represent a value which makes '
+			  + 'masks appear. Currently, no mask is displayed')
+		mask_forw = False
+		mask_backw = False
+
+'''Constants (default values and objects)'''
+#from psychopy.visual import Window, ImageStim, TextStim
+#from psychopy.core import wait
+#from psychopy.event import getKeys, clearEvents, waitKeys
+#from random import shuffle
+
+#default values
+#standard coordinates for the picture locations
+center, left, right = (0, 0), (-250, 0), (250, 0) #Mr/Ms reviewer, is this good or bad style?
```

Comment:
I have only seen this when you apply a function which returns three outputs. But I don't think it's necessarily "bad" to do this without a function. And it saves you lines of code